Injected beans not supported in NativeImageTest

It seems adding mock in ` QuarkusTest` breaks the `NativeImageTest` as it said injected field returns null.

```
[ERROR] Errors: 
[ERROR]   NativeDatasourceIT>DatasouceTest.testDeleteAllFiles:634 » NullPointer
[ERROR]   NativeDatasourceIT>DatasouceTest.testDeleteAllIOFail:735 » NullPointer
[ERROR]   NativeDatasourceIT>DatasouceTest.testDeleteFileExist:443 » NullPointer
[ERROR]   NativeDatasourceIT>DatasouceTest.testDeleteFileIOFail:563 » NullPointer
[ERROR]   NativeDatasourceIT>DatasouceTest.testDeleteFileNotExist:518 » NullPointer
[ERROR]   NativeDatasourceIT>DatasouceTest.testGetList:174 » NullPointer
[ERROR]   NativeDatasourceIT>DatasouceTest.testGetListEmpty:260 » NullPointer
[ERROR]   NativeDatasourceIT>DatasouceTest.testGetSearch:316 » NullPointer
[ERROR]   NativeDatasourceIT>DatasouceTest.testPostLoad:133 » NullPointer
[ERROR]   NativeDatasourceIT>DatasouceTest.testPostQueryTable:399 » NullPointer
[ERROR]   NativeDatasourceIT>DatasouceTest.testPostQueryTimeseries:355 » NullPointer
[ERROR]   NativeDatasourceIT>DatasouceTest.testPostSet:92 » NullPointer
[ERROR]   NativeDatasourceIT>DatasouceTest.testPostUpload:54 » NullPointer
```

I read the [documentation](https://javadoc.io/static/io.quarkus/quarkus-junit5/1.13.3.Final/io/quarkus/test/junit/NativeImageTest.html). Here is what they say.

> This also means that injection of beans into a test class using @Inject is not supported in native image tests. Such injection is only possible in tests injected with @QuarkusTest so the test class structure must take this into account.

Is there a way to work around it or should we have a separate test class `NativeDatasourceIT` for native image instead of extending the  base `DatasouceTest`?

--- a/src/test/java/io/cryostat/jfr/datasource/server/NativeDatasourceIT.java
+++ b/src/test/java/io/cryostat/jfr/datasource/server/NativeDatasourceIT.java
@@ -1,6 +1,180 @@
 package io.cryostat.jfr.datasource.server;
 
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.File;
+import java.nio.file.Files;
+
 import io.quarkus.test.junit.NativeImageTest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
 
 @NativeImageTest
-public class NativeDatasourceIT extends DatasouceTest {}
+public class NativeDatasourceIT {
+
+    @AfterEach
+    public void afterEachDatasourceTest() {
+        File directory = new File(System.getProperty("java.io.tmpdir"), "jfr-file-uploads");
+        if (directory.exists() && directory.isDirectory()) {
+            for (File f : directory.listFiles()) {
+                if (f.isFile()) {
+                    f.delete();
+                }
+            }
+        }
+        directory.delete();
+    }
+
+    @Test
+    public void testGet() throws Exception {
+        given().when().get("/").then().statusCode(200).body(is(""));
+    }
+
+    @Test
+    public void testPostUpload() throws Exception {
+        File jfrFile = new File("src/test/resources/jmc.cpu.jfr");
+        assertTrue(jfrFile.exists());
+
+        String expected = "Uploaded: jmc.cpu.jfr" + System.lineSeparator();
+        given().multiPart(jfrFile).when().post("/upload").then().statusCode(200).body(is(expected));
+    }
+
+    @Test
+    public void testPostSet() throws Exception {
+        File jfrFile = new File("src/test/resources/jmc.cpu.jfr");
+        assertTrue(jfrFile.exists());
+
+        String expected = "Uploaded: jmc.cpu.jfr" + System.lineSeparator();
+        given().multiPart(jfrFile).when().post("/upload").then().statusCode(200).body(is(expected));
+
+        expected = "Set: jmc.cpu.jfr" + System.lineSeparator();
+        given().body("jmc.cpu.jfr").when().post("/set").then().statusCode(200).body(is(expected));
+    }
+
+    @Test
+    public void testPostLoad() throws Exception {
+        File jfrFile = new File("src/test/resources/jmc.cpu.jfr");
+        assertTrue(jfrFile.exists());
+
+        String expected =
+                "Uploaded: jmc.cpu.jfr"
+                        + System.lineSeparator()
+                        + "Set: jmc.cpu.jfr"
+                        + System.lineSeparator();
+        given().multiPart(jfrFile).when().post("/load").then().statusCode(200).body(is(expected));
+    }
+
+    @Test
+    public void testGetList() throws Exception {
+        File jfrFile = new File("src/test/resources/jmc.cpu.jfr");
+        assertTrue(jfrFile.exists());
+
+        String expected = "Uploaded: jmc.cpu.jfr" + System.lineSeparator();
+        given().multiPart(jfrFile).when().post("/upload").then().statusCode(200).body(is(expected));
+
+        expected = "jmc.cpu.jfr" + System.lineSeparator();
+        given().when().get("/list").then().statusCode(200).body(is(expected));
+    }
+
+    @Test
+    public void testGetListEmpty() throws Exception {
+        String expected = "";
+        given().when().get("/list").then().statusCode(200).body(is(expected));
+    }
+
+    @Test
+    public void testGetSearch() throws Exception {
+        File jfrFile = new File("src/test/resources/jmc.cpu.jfr");
+        assertTrue(jfrFile.exists());
+
+        String expected =
+                "Uploaded: jmc.cpu.jfr"
+                        + System.lineSeparator()
+                        + "Set: jmc.cpu.jfr"
+                        + System.lineSeparator();
+        given().multiPart(jfrFile).when().post("/load").then().statusCode(200).body(is(expected));
+
+        File outputFile = new File("src/test/resources/search.output.txt");
+        assertTrue(outputFile.exists());
+
+        expected = new String(Files.readAllBytes(outputFile.toPath()));
+        given().when().get("/search").then().statusCode(200).body(is(expected));
+    }
+
+    @Test
+    public void testPostQueryTimeseries() throws Exception {
+        File jfrFile = new File("src/test/resources/jmc.cpu.jfr");
+        assertTrue(jfrFile.exists());
+
+        String expected =
+                "Uploaded: jmc.cpu.jfr"
+                        + System.lineSeparator()
+                        + "Set: jmc.cpu.jfr"
+                        + System.lineSeparator();
+        given().multiPart(jfrFile).when().post("/load").then().statusCode(200).body(is(expected));
+
+        File inputFile = new File("src/test/resources/query.timeseries.input.txt");
+        assertTrue(inputFile.exists());
+        String input = new String(Files.readAllBytes(inputFile.toPath()));
+
+        File outputFile = new File("src/test/resources/query.timeseries.output.txt");
+
+        assertTrue(outputFile.exists());
+        expected = new String(Files.readAllBytes(outputFile.toPath()));
+        given().body(input).when().post("/query").then().statusCode(200).body(is(expected));
+    }
+
+    @Test
+    public void testPostQueryTable() throws Exception {
+        File jfrFile = new File("src/test/resources/jmc.cpu.jfr");
+        assertTrue(jfrFile.exists());
+
+        String expected =
+                "Uploaded: jmc.cpu.jfr"
+                        + System.lineSeparator()
+                        + "Set: jmc.cpu.jfr"
+                        + System.lineSeparator();
+        given().multiPart(jfrFile).when().post("/load").then().statusCode(200).body(is(expected));
+
+        File inputFile = new File("src/test/resources/query.table.input.txt");
+        assertTrue(inputFile.exists());
+        String input = new String(Files.readAllBytes(inputFile.toPath()));
+
+        File outputFile = new File("src/test/resources/query.table.output.txt");
+
+        assertTrue(outputFile.exists());
+        expected = new String(Files.readAllBytes(outputFile.toPath()));
+        given().body(input).when().post("/query").then().statusCode(200).body(is(expected));
+    }
+
+    @Test
+    public void testDeleteFileExist() throws Exception {
+        File jfrFile = new File("src/test/resources/jmc.cpu.jfr");
+        assertTrue(jfrFile.exists());
+
+        String expected = "Uploaded: jmc.cpu.jfr" + System.lineSeparator();
+        given().multiPart(jfrFile).when().post("/upload").then().statusCode(200).body(is(expected));
+
+        given().body("jmc.cpu.jfr").when().delete("/delete").then().statusCode(204).body(is(""));
+    }
+
+    @Test
+    public void testDeleteFileNotExist() throws Exception {
+        given().body("jmc.cpu.jfr").when().delete("/delete").then().statusCode(404).body(is(""));
+    }
+
+    @Test
+    public void testDeleteAllFiles() throws Exception {
+        File jfrFile = new File("src/test/resources/jmc.cpu.jfr");
+        assertTrue(jfrFile.exists());
+
+        String expected = "Uploaded: jmc.cpu.jfr" + System.lineSeparator();
+        given().multiPart(jfrFile).when().post("/upload").then().statusCode(200).body(is(expected));
+
+        expected = "Deleted: jmc.cpu.jfr" + System.lineSeparator();
+        given().when().delete("/delete_all").then().statusCode(200).body(is(expected));
+        given().when().delete("/delete_all").then().statusCode(200).body(is(""));
+    }
+}

